internal/settings/settings.go

```diff
@@ -119,3 +120,44 @@ func (s *Settings) SetUsername(username string) {
 func (s *Settings) GetUsername() string {
 	return viper.GetString("username")
 }
+
+func (s *Settings) SetAutoupdate(autoupdate string) {
+	config := viper.GetStringMap("config")
+	if config == nil {
+		config = make(map[string]interface{})
+	}
+	config["autoupdate"] = autoupdate
+	viper.Set("config", config)
+	s.changed = true
+}
+
+func (s *Settings) SetLastUpdateCheck(t int64) {
+	config := viper.GetStringMap("config")
+	if config == nil {
+		config = make(map[string]interface{})
+	}
+	config["last_update_check"] = t
+	viper.Set("config", config)
+	s.changed = true
+}
+
+func (s *Settings) GetLastUpdateCheck() int64 {
+	config := viper.GetStringMap("config")
+	if config == nil {
+		return 0
+	}
+	lastUpdateCheck, ok := config["last_update_check"]
+	if !ok {
+		return 0
+	}
+	return lastUpdateCheck.(int64)
+}
+
+func (s *Settings) GetAutoupdate() string {
+	config := viper.GetStringMap("config")
+	if config == nil {
+		return "on"
+	}
+	value := config["autoupdate"]
```

Comment:
The value itself could also be empty. We should probably return `on` when that happens as well.